pyccel/parser/semantic.py

```diff
@@ -344,17 +344,39 @@ def change_to_module_scope(self):
 
     def check_for_variable(self, name):
         """
+        Search for a Variable object with the given name in the current scope.
+
         Search for a Variable object with the given name in the current scope,
         defined by the local and global Python scopes. Return None if not found.
+
+        Parameters
+        ----------
+        name : str
+            The object describing the variable.
+
+        Returns
+        -------
+        Variable
+            Returns the varibale if found or None.
         """
 
-        if self.current_class:
-            for i in self._current_class.attributes:
-                if i.name == name:
-                    var = i
-                    return var
-            return None
+        if isinstance(name, DottedName):
+            prefix_parts = name.name[:-1]
+            syntactic_prefix = prefix_parts[0] if len(prefix_parts) == 1 else DottedName(*prefix_parts)
+            prefix = self._visit(syntactic_prefix)
+            class_def = prefix.cls_base
+            attr_name = name.name[-1]
+            attribute = class_def.scope.find(attr_name, 'variables')
+            if attribute:
+                return attribute
+            else:
+                return None
         else:
+            if self.current_class:
+                for i in self._current_class.attributes:
+                    if i.name == name:
+                        var = i
+                        return var
```

Comment:
Can you remind me why this is necessary please. Is it because `_visit_DottedName` strips off the prefix? Can `_visit_DottedName` pass the full `DottedName` instead? If not, why not?

Reply:
This code is used for : https://github.com/pyccel/pyccel/blob/e7b19d20720e2c2800a7858162928c0aae6ea639/pyccel/parser/semantic.py#L2125-L2128
You suggested to change it to a simple `second = self.check_for_variable(rhs_new_name)`.
I couldn't find a way to pass it from : https://github.com/pyccel/pyccel/blob/e7b19d20720e2c2800a7858162928c0aae6ea639/pyccel/parser/semantic.py#L363-L369

Reply:
No. I suggest to change this:
https://github.com/pyccel/pyccel/blob/e7b19d20720e2c2800a7858162928c0aae6ea639/pyccel/parser/semantic.py#L2123-L2129
to:
```python
            # standard class attribute 
             second = self.check_for_variable(expr)
             if second: 
                 return second
```

Reply:
We can't do that cause for exemple if we have : `get_Point_Class().x`; `get__Point_Class()` is a `FunctionCall` and it doesn't have a `cls_base`attributes so it will return an Error in `check_for_variable()`.

Reply:
Ah ok makes sense. I wonder if `cls_base` should be added to `PyccelAstNode` at some point in the future. You could also use the `dtype` and the scope to get the class def

Reply:
Fixed https://github.com/pyccel/pyccel/pull/1484/commits/d0556e56b54da503b742820df96af7ad95b0c520. 
I left the `second.clone()` because we need it to be a `DottedVariable`.

Reply:
Do you think `check_for_variable` should return a DottedVariable? That makes more sense to me

Reply:
Fixed https://github.com/pyccel/pyccel/pull/1484/commits/f88f28431dfd69ed4666bba5378025beb7c455ea.